src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,92 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_SIZE = 10;
+    private static final double MULTIPLIER = 1.5;
+    private int size;
+    private Object[] array;
+
+    public ArrayList() {
+        array = new Object[MAX_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        resize();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index, size + 1);
+        resize();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, size);
+        return (T) array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index, size);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, size);
+        T removedElement = (T) array[index];
+        System.arraycopy(array, index + 1, array, index, size - index - 1);
+        size--;
+        return removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (array[i] == element
+                    || (array[i] != null) && array[i].equals(element)) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("No such element");
```

Comment:
you need to write more informative messages - indicate with which index/element the error occurred. Corrections here and in the following cases
```suggestion
        throw new NoSuchElementException("No such element " + element);
```